openhands/sdk/conversation/title_utils.py

```diff
@@ -0,0 +1,172 @@
+"""Utility functions for generating conversation titles."""
+
+from typing import Optional
+
+from litellm.types.utils import Choices
+
+from openhands.sdk.event import MessageEvent
+from openhands.sdk.llm import LLM, Message, TextContent
+from openhands.sdk.logger import get_logger
+from openhands.sdk.utils.protocol import ListLike
+
+
+logger = get_logger(__name__)
+
+
+def extract_first_user_message(events: ListLike) -> Optional[str]:
+    """Extract the first user message from conversation events.
+
+    Args:
+        events: List of conversation events.
+
+    Returns:
+        The first user message text, or None if no user message is found.
+    """
+    for event in events:
+        if (
+            isinstance(event, MessageEvent)
+            and event.source == "user"
+            and event.llm_message.content
+        ):
+            # Extract text content from the message
+            text_parts = []
+            for content in event.llm_message.content:
+                if isinstance(content, TextContent):
+                    text_parts.append(content.text)
+
+            if text_parts:
+                return " ".join(text_parts).strip()
+
+    return None
+
+
+def generate_title_with_llm(
+    message: str, llm: LLM, max_length: int = 50
+) -> Optional[str]:
+    """Generate a conversation title using LLM.
+
+    Args:
+        message: The first user message to generate title from.
+        llm: The LLM to use for title generation.
+        max_length: Maximum length of the generated title.
+
+    Returns:
+        Generated title, or None if LLM fails or returns empty response.
+    """
+    # Truncate very long messages to avoid excessive token usage
+    if len(message) > 1000:
+        truncated_message = message[:1000] + "...(truncated)"
+    else:
+        truncated_message = message
+
+    try:
+        # Create messages for the LLM to generate a title
+        messages = [
+            Message(
+                role="system",
+                content=[
+                    TextContent(
+                        text=(
+                            "You are a helpful assistant that generates concise, "
+                            "descriptive titles for conversations with OpenHands. "
+                            "OpenHands is a helpful AI agent that can interact "
+                            "with a computer to solve tasks using bash terminal, "
+                            "file editor, and browser. Given a user message "
+                            "(which may be truncated), generate a concise, "
+                            "descriptive title for the conversation. Return only "
+                            "the title, with no additional text, quotes, or "
+                            "explanations."
```

Comment:
Read https://gitmoji.dev/ and come up with a list of rules that conversation title should starts with emoji.

For example, We could define a vocabulary like:
:art: Frontend
:gear: Backend
:test_tube: Tests
:zap: DevOps
:rocket: Deployment
:package: Dependencies
:cd: Database
:broom: Chores


But please refer to https://gitmoji.dev/ for a list of standardized emojis